### PR TITLE
[WFL] deal with formula_callable inputs that currently don't do anything

### PR DESCRIPTION
**Pull request by soliton-**
The "undead" attribute for `unit_type_callable` was split in https://github.com/wesnoth/wesnoth/commit/d82762b87d8#diff-7842222d67a8d773accb3fa2e9b89c54e5342a648566c987c88004eb8998b1bc but the input remained and the split was not done for `unit_callable`.

For 1.18 I think it might be best to fix the input to have it work for unit and unit type. It can be removed for 1.20.

### CODE REVIEW

File: src/formula/callable_objects.cpp [hunk 1]
```diff
@@ -750,6 +752,8 @@ variant team_callable::get_value(const std::string& key) const
 		return variant(team_.village_support());
 	} else if(key == "recall_cost") {
 		return variant(team_.recall_cost());
+	} else if(key == "name") {
+		return variant(team_.current_player());
```
[the remark this answers is not in the document]
Comment (soliton-) [reply src/formula/callable_objects.cpp:756]: It was removed in https://github.com/wesnoth/wesnoth/commit/31ae13da1eee59eb12913c05f786f7ea361206a9#diff-5528f94fe46e41d370e827696891a90033fd8e07b3bf337f7255250929ac81cc.

I didn't check what it contained then.

We can also just remove the input.

**Comment by soliton-** [comment]
The alternative is to just remove the inputs that don't do anything.

I think it's a bit confusing in the case of unit vs unit type but it probably doesn't matter that much anyway. I think the inputs might just get used to print stuff when using the formula console.

**Comment by soliton-** [comment]
Should have made it clearer from the start that this is more of an open question on how best to deal with out of sync `get_inputs` and `get_value` implementations of `formula_callable` derived classes. Mostly regarding specified inputs that don't have any implementation in `get_value` the other way around is probably more intentional.

**Comment by soliton-** [comment]
I've looked through all formula_callable classes now and found no further cases.

**Comment by CelticMinstrel** [comment]
I don't really have any complaints about this.

File: src/formula/callable_objects.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (stevecotton) [reply src/formula/callable_objects.cpp:756]: I think I lean towards removing it, but either way I'm OK with this PR.

**Comment by stevecotton** [comment]
Are we waiting for anything before merging?

**Comment by soliton-** [comment]
I wanted to go with removing "name" instead. I probably won’t manage that before next year though.

**Pull request merged**